tests/bluetooth/audio/ascs/src/test_ase_state_transition_invalid.c

```diff
@@ -40,18 +41,27 @@ static void *test_ase_state_transition_invalid_setup(void)
 	fixture = malloc(sizeof(*fixture));
 	zassert_not_null(fixture);
 
-	memset(fixture, 0, sizeof(*fixture));
-	fixture->ase_cp = test_ase_control_point_get();
-	test_conn_init(&fixture->conn);
-	test_ase_snk_get(1, &fixture->ase_snk);
-	test_ase_src_get(1, &fixture->ase_src);
 
 	return fixture;
 }
 
 static void test_ase_state_transition_invalid_before(void *f)
 {
+	struct test_ase_state_transition_invalid_fixture *fixture =
+		(struct test_ase_state_transition_invalid_fixture *)f;
+	struct bt_bap_unicast_server_register_param param = {
+		CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT,
+		CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT
+	};
+
+	bt_bap_unicast_server_register(&param);
 	bt_bap_unicast_server_register_cb(&mock_bap_unicast_server_cb);
+
+	memset(fixture, 0, sizeof(struct test_ase_state_transition_invalid_fixture));
+	fixture->ase_cp = test_ase_control_point_get();
+	test_conn_init(&fixture->conn);
+	test_ase_snk_get(1, &fixture->ase_snk);
+	test_ase_src_get(1, &fixture->ase_src);
```

Comment:
Shouldn't this be
```suggestion
	test_ase_snk_get(CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT, &fixture->ase_snk);
	test_ase_src_get(CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT, &fixture->ase_src);
```

So that it matches the number of ASEs we register?

We should also consider verifying that it actually works when we register fewer ASEs than the max

Reply:
> We should also consider verifying that it actually works when we register fewer ASEs than the max

This is already done in test 'test_ascs_register_zero_ases'. Do you mean even further testing?
I do think that maybe a simulation test should be written for this as well, where changes to num of ases are made on run time. But as Im sure you can guess, I'd prefer to not add that in this commit due to its already big size.

Reply:
> This is already done in test 'test_ascs_register_zero_ases'. Do you mean even further testing?

Yes, if `CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT=2` then we only test for `0`, `CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT (2)` and `CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT + 1 (3)`, but not `1`.

In other words, we actually never test that one of the main feature added by this PR (registering something smaller than `CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT`) is actually working

Reply:
Actually you are very right about this.

I would then however need to bump the number of ASES from 1 to 2 globally for these test cases - and I was a bit afraid that this will cause them to break down :D